azure-pipelines.yml

```diff
@@ -334,3 +338,70 @@ jobs:
       packagesToPush: '$(OutputPackagesDir)/**/*.nupkg;!$(OutputPackagesDir)/**/*.symbols.nupkg'
       publishVstsFeed: 'c1408dcb-1833-4ae4-9af5-1a891a12cc3c'
       allowPackageConflicts: true
+
+- job: build_docs
+  displayName: Build API docs
+  pool:
+    vmImage: ubuntu-20.04
+  steps:
+  - checkout: self
+    clean: true
+    submodules: recursive
+  - task: UseDotNet@2
+    displayName: ⚙ Install .NET SDK
+    inputs:
+      packageType: sdk
+      useGlobalJson: true
+      workingDirectory: apidocs
+
+  # ESRP Authenticode sign package DLLs
+  - task: UseDotNet@2
+    displayName: ⚙ Install .NET Core 2.1.x
+    inputs:
+      packageType: runtime
+      version: 2.1.x
+
+  - script: dotnet pack -c $(BuildConfiguration)
+    displayName: 📦 dotnet pack
+    workingDirectory: apidocs
+  - task: EsrpCodeSigning@1
+    displayName: ✒ NuGet sign
+    inputs:
+      ConnectedServiceName: Undocked RegFree Signing Connection
+      FolderPath: $(System.DefaultWorkingDirectory)/bin/Packages/$(BuildConfiguration)/NuGet
+      Pattern: '*.nupkg'
+      signConfigType: inlineSignParams
+      inlineOperation: |
+        [
+          {
+            "KeyCode" : "CP-401405",
+            "OperationCode" : "NuGetSign",
+            "Parameters" : {},
+            "ToolName" : "sign",
+            "ToolVersion" : "1.0"
+          },
+          {
+            "KeyCode" : "CP-401405",
+            "OperationCode" : "NuGetVerify",
+            "Parameters" : {},
+            "ToolName" : "sign",
+            "ToolVersion" : "1.0"
+          }
+        ]
+      SessionTimeout: 60
+      MaxConcurrency: 50
+      MaxRetryAttempts: 5
+    condition: and(succeeded(), eq(variables['SignFiles'], 'true'))
+  - publish: bin/Packages/$(BuildConfiguration)/NuGet
+    artifact: ApiDocsNuGetPackages
+    displayName: 📢 Publish package
+    # There's a problem on microsoft.visualstudio.com that requires the guid instead of NuGetCommand@2
+    # Don't publish if we're using pre-generated source
+  - task: 333b11bd-d341-40d9-afcf-b32d5ce6f23b@2
+    displayName: 📤 NuGet push
+    inputs:
+      command: push
+      packagesToPush: $(System.DefaultWorkingDirectory)/bin/Packages/$(BuildConfiguration)/NuGet/*.nupkg
+      publishVstsFeed: c1408dcb-1833-4ae4-9af5-1a891a12cc3c
+      allowPackageConflicts: true
+    condition: and(succeeded(), ne(variables['Build.Reason'], 'PullRequest'))
```

Comment:
I'm *not* pushing packages on PR. But other jobs in this pipeline do. I would suggest as a separate PR that we avoid pushing packages anywhere on PR.
I'd also like to refactor this file to allow for publicly observable pipelines on `azure-public` so our OSS contributors can access them, the way we have for CsWin32 today.